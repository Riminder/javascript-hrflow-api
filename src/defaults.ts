let defaults: any = {
  API_URL: "https://api.hrflow.ai/v1",
  API_SECRET: null,
  api_secret: null,
  api_user: null,
};

export default defaults;