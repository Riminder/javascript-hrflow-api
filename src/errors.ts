import { HrflowAPIResponse } from "./types";

export class APIError extends Error {
  response: HrflowAPIResponse;
  constructor(message: string, response: HrflowAPIResponse)Â {
    super(message);
    this.response = response;
  }
}