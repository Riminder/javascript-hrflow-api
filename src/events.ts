export default [
  "profile.parsing.success",
  "profile.parsing.error",
  "profile.parsing.update",
  "profile.scoring.success",
  "profile.scoring.error",
  "profile.scoring.update",
  "profile.revealing.success",
  "profile.revealing.error",
  "profile.revealing.update",
  "profile.reasoning.success",
  "profile.reasoning.error",
  "profile.reasoning.update",
  "profile.embedding.success",
  "profile.embedding.error",
  "profile.embedding.update",
  "job.parsing.success",
  "job.parsing.error",
  "job.parsing.update",
  "job.scoring.success",
  "job.scoring.error",
  "job.scoring.update",
  "job.revealing.success",
  "job.revealing.error",
  "job.revealing.update",
  "job.reasoning.success",
  "job.reasoning.error",
  "job.reasoning.update",
  "job.embedding.success",
  "job.embedding.error",
  "job.embedding.update",
  "action.stage.success",
  "action.stage.error",
  "action.rating.success",
  "action.rating.error",
  "agent.training.success",
  "agent.training.error",
  "agent.training.update",
  "agent.scoring.success",
  "agent.scoring.error",
  "agent.scoring.update",

];