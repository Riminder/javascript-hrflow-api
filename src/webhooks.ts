import Hrflow = require("./index");
import Events from "./events";
import * as util from "tweetnacl-util";
import * as sha256 from "fast-sha256";
import { WebhooksResponse } from "./types";
import defaults from "./defaults";
import { httpPostRequest } from "./http";

export namespace Webhooks {
  export interface ResponseBase {
    type: string;
    message: string;
  }
  export interface ProfileParseResponse extends ResponseBase {
    profile: Profile;
  }

  export interface ProfileScoreResponse extends ResponseBase {
    profile: Profile;
    filter: Filter;
    score: number;
  }

  export interface FilterTrainResponse extends ResponseBase {
    filter: Filter;
  }

  export interface FilterScoreResponse extends ResponseBase {
    filter: Filter;
  }

  export interface ActionStageResponse extends ResponseBase {
    profile: Profile;
    filter: Filter;
    stage: string;
  }

  export interface ActionRatingResponse extends ResponseBase {
    profile: Profile;
    filter: Filter;
    rating: number;
  }

  export type Response = ProfileParseResponse |
    ProfileScoreResponse |
    FilterTrainResponse |
    FilterScoreResponse |
    ActionRatingResponse |
    ActionStageResponse;

  export interface Profile {
    profile_key: string;
    profile_reference: string;
  }

  export interface Filter {
    filter_id: string;
    filter_reference: string;
  }

  export type EventCallbackMap = Map<string, (data: Webhooks.Response, type: string) => any>;
}

export class Webhooks {
  private hrflow: Hrflow;
  binding: Webhooks.EventCallbackMap;

  constructor(hrflow: Hrflow) {
    if (!hrflow  || !hrflow.Webhooks_Key) {
      throw new Error("The webhook secret key must be specified");
    }

    this.hrflow = hrflow;
    this.binding = new Map<string, (data: Webhooks.Response, type: string) => any>();
  }

  handle(headers: any): () => void {
    return () => {
      if (!headers["HTTP-RIMINDER-SIGNATURE"]) {
        throw new Error("The signature is missing from the headers");
      }

      const [encodedSignature, encodedPayload] = headers["HTTP-RIMINDER-SIGNATURE"].split(".");
      const expectedSignature = util.encodeBase64(sha256.hmac(util.decodeUTF8(this.hrflow.Webhooks_Key), util.decodeUTF8(encodedPayload)));

      if (encodedSignature !== expectedSignature) {
        throw new Error("The signature is invalid");
      }

      const payload: Webhooks.Response = JSON.parse(util.encodeUTF8(util.decodeBase64(encodedPayload)));

      if (Events.indexOf(payload.type) < 0) {
        throw new Error(`Unknown event: ${payload.type}`);
      }
      this._callBinding(payload);
    };
  }

  on(event: string, callback: (data: Webhooks.Response, type?: string) => any) {
    if (Events.indexOf(event) < 0) {
      throw new Error("This event doesn't exist");
    }

    if (this.binding.has(event)) {
      throw new Error("This callback already has been declared");
    }

    this.binding.set(event, callback);

    return this;
  }

  check(): Promise<WebhooksResponse> {
    const url = `${defaults.API_URL}/webhook/check`;
    return httpPostRequest(url, null, { headers: this.hrflow.headers });
  }

  private _callBinding(payload: Webhooks.Response): void {
    if (this.binding.has(payload.type)) {
      this.binding.get(payload.type)(payload, payload.type);
    }
  }
}