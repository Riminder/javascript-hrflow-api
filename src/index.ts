import { HrflowOptions } from "./types";
import { Webhooks } from "./webhooks";
import Source from "./resources/source";
import Profile from "./resources/profile/profile";
import Offer from "./resources/job/job";

class Hrflow {
  public headers: any;
  public API_Key: string;
  public Webhooks_Key: string;
  public webhooks: Webhooks;
  public source: Source;
  public profile: Profile;
  public offer: Offer;

  constructor(options: HrflowOptions) {

    if (!options.API_Key) {
      let error = new Error("No API Key was supplied for Hrflow SDK");
      throw error;
    }

    this.API_Key = options.API_Key;
    this.headers = {
      "X-API-Key": this.API_Key,
    };
    if (options.Webhooks_Key) {
      this.Webhooks_Key = options.Webhooks_Key;
    }

    this._init();
  }

  private _init() {
    if (this.Webhooks_Key) {
      this.webhooks = new Webhooks(this);
    }

    this.source = new Source(this);
    this.profile = new Profile(this);
    this.offer = new Offer(this);
  }
}

export = Hrflow;