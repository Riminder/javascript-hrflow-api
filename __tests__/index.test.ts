import * as fs from "fs";
import { generateURLParams } from "../src/utils";
import { httpRequest } from "../src/http";
import Hrflow = require("../src/index");
import { HrflowOptions, profilesSearchingOptions, ProfileUpload, Stage, SortBy, OrderBy, ProfileJSON } from "../src/types";

let app: Hrflow;

/**
 * @Readme: These tests are incomplete. We should also test the content of the request. It's currently testing
 * only the endpoint and the query parameters given.
 * @Todo: Add body check for each post/patch route
 */

function getQueryParamAsArray(query: string): Array<string> {
  return query
    .split("&")
    .map((entry: string) => entry.split("=")[0]);
}

describe("Other tests", () => {
  describe("\"Hrflow\" object relative tests", () => {
    test("it should throw an error when no key is given", () => {
      expect(() => {
        app = new Hrflow({} as HrflowOptions);
      }).toThrowErrorMatchingSnapshot();
    });
  });

  describe("utils module relative tests", () => {
    test("This should return null if there is no input data", () => {
      expect(generateURLParams(null)).toBeNull();
    });
  });

  describe("http module relative tests", () => {
    test("It should throw an error if the error code is not 200 or 201", () => {
      const headers = {
        "X-API-KEY": "api_key",
      };

      expect.assertions(1);
      return httpRequest("localhost", { error: true, headers }).catch((e) => {
        expect(e).toMatchSnapshot();
      });
    });

    test("it should throw an error if something went wrong with the request", () => {
      const headers = {
        "X-API-KEY": "api_key",
      };
      expect.assertions(1);
      return httpRequest("localhost", { reject: true, headers }).catch((e) => {
        expect(e).toMatchSnapshot();
      });
    });
  });
});

describe("Wrapper test", () => {
    app = new Hrflow({api_secret : "api_key", api_user: "api_user"});
    describe("Source endpoints", () => {
        test("It should call the get source list endpoint", () => {
          const options = {};
          app.source.list(options).then((response: any) => {
            expect(response).toMatchSnapshot();
          })
        }
       );

        test("It should call the get source endpoint", () => {
          app.source.get("source_key").then((response: any) => {
            expect(response).toMatchSnapshot();
          });
        });
      });

    describe("Profile endpoints", () => {
      test("It should call the get profile endpoint using Date object", () => {
        const options: profilesSearchingOptions = {
          source_keys: ["source1", "source2"],
          created_at_min: new Date(0),
          created_at_max: new Date(1234),
          page: 1,
          stage: Stage.YES,
          limit: 30,
          sort_by: SortBy.DATE_RECEPTION,
          order_by: OrderBy.DESC,
        };
        app.profile.searching.list(options)
          .then((response: any) => {
          expect(response).toMatchSnapshot();
          expect(getQueryParamAsArray(response.url.query)).toEqual(Object.keys(options));
        });
      });

      test("It should call the get profile endpoint using Date number", () => {
        const options: profilesSearchingOptions = {
          source_keys: ["source1", "source2"],
          created_at_min: 0,
          created_at_max: (new Date("2018-01-01")).getTime(),
          page: 1,
          stage: Stage.YES,
          limit: 30,
          sort_by: SortBy.DATE_RECEPTION,
          order_by: OrderBy.DESC,
        };
        app.profile.searching.list(options)
          .then((response: any) => {
          expect(response).toMatchSnapshot();
          expect(getQueryParamAsArray(response.url.query)).toEqual(Object.keys(options));
        });
      });

      test("It should call the get profile endpoint with the profile id", () => {
        const options = {
          source_key: "source_key",
          profile_key: "profile_key",
        };
        app.profile.get(options).then((response: any) => {
          expect(response).toMatchSnapshot();
        });
      });

      test("It should call the get profile endpoint with the profile reference", () => {
        const options = {
          source_key: "source_key",
          reference: "reference",
        };
        app.profile.get(options).then((response: any) => {
          expect(response).toMatchSnapshot();
        });
      });

      test("It should call the get profile attachments endpoint with the profile id", () => {
        const options = {
          source_key: "source_key",
          profile_key: "profile_key",
        };
        app.profile.attachments.list(options).then((response: any) => {
          expect(response).toMatchSnapshot();
        });
      });

      test("It should call the get profile document endpoint with the profile reference", () => {
        const options = {
          source_key: "source_key",
          reference: "reference",
        };
        app.profile.attachments.list(options).then((response: any) => {
          expect(response).toMatchSnapshot();
        });
      });

      test("It should call the get profile parsing endpoint with the profile id", () => {
        const options = {
          source_key: "source_key",
          profile_key: "profile_key",
        };
        app.profile.parsing.get(options).then((response: any) => {
          expect(response).toMatchSnapshot();
        });
      });

      test("It should call the get profile parsing endpoint with the profile reference", () => {
        const options = {
          source_key: "source_key",
          reference: "reference",
        };
        app.profile.parsing.get(options).then((response: any) => {
          expect(response).toMatchSnapshot();
        });
      });

      test("It should call the get profile scoring endpoint with the profile id", () => {
        const options = {
          source_keys:  ["source_key"],
          job_key: "job_key",
          board_key: "board_key",
        };
        app.profile.scoring.list(options).then((response: any) => {
          expect(response).toMatchSnapshot();
        });
      });

      test("It should call the get profile scoring endpoint with the profile reference", () => {
        const options = {
          source_keys:  ["source_key"],
          job_key: "job_key",
          board_key: "board_key",
        };
        app.profile.scoring.list(options).then((response: any) => {
          expect(response).toMatchSnapshot();
        });
      });

      test("It should call the post resume endpoint", () => {
        const file = fs.createReadStream("./test.txt");
        const data: ProfileUpload = {
          source_key: "source_key",
          file: file,
          reference: "ref",
          created_at: new Date(Date.now()),
          metadatas: [{
            name: "filter_reference",
            value: "123456"
          }],
        };

        app.profile.addFile(data)
          .then((response: any) => {
            const responseWithoutBody = {
              url: response.url,
              options: {
                headers: response.options.headers,
                method: response.options.method
              }
            };
            expect(responseWithoutBody).toMatchSnapshot();
        });
      });

      // test("It should call the patch stage endpoint with the the profile id and filter id", () => {
      //   const data: StagePatch = {
      //     source_key: "source_key",
      //     profile_key: "profile_key",
      //     filter_id: "filter_id",
      //     stage: Stage.YES,
      //   };

      //   app.profile.stage.set(data)
      //     .then((response: any) => {
      //       const responseWithoutBody = {
      //         url: response.url,
      //         options: {
      //           headers: response.options.headers,
      //           method: response.options.method
      //         }
      //       };
      //       expect(responseWithoutBody).toMatchSnapshot();
      //     });
      // });

      // test("It should call the patch stage endpoint with the the profile id and filter reference", () => {
      //   const data: StagePatch = {
      //     source_key: "source_key",
      //     profile_key: "profile_key",
      //     filter_id: "filter_id",
      //     stage: Stage.YES,
      //   };
      //   app.profile.stage.set(data)
      //     .then((response: any) => {
      //       const responseWithoutBody = {
      //         url: response.url,
      //         options: {
      //           headers: response.options.headers,
      //           method: response.options.method
      //         }
      //       };
      //       expect(responseWithoutBody).toMatchSnapshot();
      //     });
      // });

      // test("It should call the patch stage endpoint with the the profile reference and filter id", () => {
      //   const data: StagePatch = {
      //     source_key: "source_key",
      //     profile_key: "profile_key",
      //     filter_id: "filter_id",
      //     stage: Stage.YES,
      //   };
      //   app.profile.stage.set(data)
      //     .then((response: any) => {
      //       const responseWithoutBody = {
      //         url: response.url,
      //         options: {
      //           headers: response.options.headers,
      //           method: response.options.method
      //         }
      //       };
      //       expect(responseWithoutBody).toMatchSnapshot();
      //     });
      // });

      // test("It should call the patch stage endpoint with the the profile reference and filter reference", () => {
      //   const data: StagePatch = {
      //     source_key: "source_key",
      //     profile_key: "profile_key",
      //     filter_id: "filter_id",
      //     stage: Stage.YES,
      //   };
      //   app.profile.stage.set(data)
      //     .then((response: any) => {
      //       const responseWithoutBody = {
      //         url: response.url,
      //         options: {
      //           headers: response.options.headers,
      //           method: response.options.method
      //         }
      //       };
      //       expect(responseWithoutBody).toMatchSnapshot();
      //     });
      // });

      // test("It should call the patch rating endpoint with the the profile id and filter id", () => {
      //   const data: RatingPatch = {
      //     source_key: "source_key",
      //     profile_key: "profile_key",
      //     filter_id: "filter_id",
      //     rating: 2
      //   };
      //   app.profile.rating.set(data)
      //     .then((response: any) => {
      //       const responseWithoutBody = {
      //         url: response.url,
      //         options: {
      //           headers: response.options.headers,
      //           method: response.options.method
      //         }
      //       };
      //       expect(responseWithoutBody).toMatchSnapshot();
      //     });
      // });

      // test("It should call the patch rating endpoint with the the profile id and filter reference", () => {
      //   const data: RatingPatch = {
      //     source_key: "source_key",
      //     profile_key: "profile_key",
      //     filter_id: "filter_id",
      //     rating: 2
      //   };
      //   app.profile.rating.set(data)
      //     .then((response: any) => {
      //       const responseWithoutBody = {
      //         url: response.url,
      //         options: {
      //           headers: response.options.headers,
      //           method: response.options.method
      //         }
      //       };
      //       expect(responseWithoutBody).toMatchSnapshot();
      //     });
      // });

      // test("It should call the patch rating endpoint with the the profile reference and filter id", () => {
      //   const data: RatingPatch = {
      //     source_key: "source_key",
      //     profile_key: "profile_key",
      //     filter_id: "filter_id",
      //     rating: 2
      //   };
      //   app.profile.rating.set(data)
      //     .then((response: any) => {
      //       const responseWithoutBody = {
      //         url: response.url,
      //         options: {
      //           headers: response.options.headers,
      //           method: response.options.method
      //         }
      //       };
      //       expect(responseWithoutBody).toMatchSnapshot();
      //     });
      // });

      // test("It should call the patch rating endpoint with the the profile reference and filter reference", () => {
      //   const data: RatingPatch = {
      //     source_key: "source_key",
      //     profile_key: "profile_key",
      //     filter_id: "filter_id",
      //     rating: 2
      //   };
      //   app.profile.rating.set(data)
      //     .then((response: any) => {
      //       const responseWithoutBody = {
      //         url: response.url,
      //         options: {
      //           headers: response.options.headers,
      //           method: response.options.method
      //         }
      //       };
      //       expect(responseWithoutBody).toMatchSnapshot();
      //     });
      // });

      test("It should call the post profile data endpoint", () => {
        const json: ProfileJSON = {
          source_key: "source_key",
          reference: "macfly",
          info: {
            full_name: "Marty McFly",
            first_name: "Marty McFly",
            last_name: "Marty McFly",
            email: "marty.mcfly@gmail.com",
            phone: "202-555-0141",
            summary: "High school student, loves to time travel",
            location: {
              text: "9303 Lyon Drive, Lyon Estates, Hill Valley CA 95420"
            },
            urls: {
              from_resume: [
                "test.com"
              ],
              linkedin: "",
              twitter: "",
              facebook: "",
              github: "",
              picture: ""
            }
          },
          experiences: [{
            date_start: "01/01/2017",
            date_end: "01/01/2018",
            title: "CusCo employee",
            company: "CusCo",
            location: {
              text: "Hill Valley"
            },
            description: "Fujitsu company"
          }],
          educations: [{
            date_start: "01/01/1985",
            date_end: "01/01/1986",
            title: "Hill Valley High School",
            school: "Hill Valley High School",
            location: {
              text: "Hill Valley"
            },
            description: "a school"
          }],
          skills: [
           {
             name: 'python',
             value: 'senior'
           }
          ],
          languages: [
            {
              name: 'english',
              value: 'fluent'
            }
          ],
          interests: [
            {
              name: 'aptitude',
              value: true,
            }
          ],
          labels: {
            job_key: "848491764713d6a6dd97480f865ffdb51077a6fb",
            job_reference: "azerty123",
            stage: "later",
            stage_timestamp: 1594806495285,
            rating: null,
            rating_timestamp: null,
          }
        };

        app.profile.addJson(json).then((response: any) => {
          const responseWithoutBody = {
            url: response.url,
            options: {
              headers: response.options.headers,
              method: response.options.method
            }
          };
          expect(responseWithoutBody).toMatchSnapshot();
        });
      });

      // test("It should call the check profile data endpoint", () => {
      //   const json: JsonUploadCheck = {
      //     profile_json: {
      //       name: "Marty McFly",
      //       email: "marty.mcfly@gmail.com",
      //       phone: "202-555-0141",
      //       summary: "High school student, loves to time travel",
      //       timestamp_reception: new Date("1985-10-21"),
      //       location_details: {
      //         text: "9303 Lyon Drive, Lyon Estates, Hill Valley CA 95420"
      //       },
      //       experiences: [{
      //         start: "01/01/2017",
      //         end: "01/01/2018",
      //         title: "CusCo employee",
      //         company: "CusCo",
      //         location_details: {
      //           text: "Hill Valley"
      //         },
      //         location: "Hill Valley",
      //         description: "Fujitsu company"
      //       }],
      //       educations: [{
      //         start: "01/01/1985",
      //         end: "01/01/1986",
      //         title: "Hill Valley High School",
      //         school: "Hill Valley High School",
      //         location_details: {
      //           text: "Hill Valley"
      //         },
      //         location: "Hill Valley",
      //         description: "a school"
      //       }],
      //       skills: [
      //         "skate",
      //         "time travel"
      //       ],
      //       languages: [
      //         "english"
      //       ],
      //       interests: [
      //         "music",
      //       ],
      //       urls: {
      //         from_resume: [
      //           "test.com"
      //         ],
      //         linkedin: "",
      //         twitter: "",
      //         facebook: "",
      //         github: "",
      //         picture: ""
      //       }
      //     }
      //   };

      //   app.profile.json.check(json).then((response: any) => {
      //     const responseWithoutBody = {
      //       url: response.url,
      //       options: {
      //         headers: response.options.headers,
      //         method: response.options.method
      //       }
      //     };
      //     expect(responseWithoutBody).toMatchSnapshot();
      //   });
      // });
    });
});