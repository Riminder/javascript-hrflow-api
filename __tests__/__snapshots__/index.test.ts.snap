// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Other tests "Hrflow" object relative tests it should throw an error when no key is given 1`] = `"No API Key was supplied for Hrflow SDK"`;

exports[`Other tests http module relative tests It should throw an error if the error code is not 200 or 201 1`] = `[Error: An error occured]`;

exports[`Other tests http module relative tests it should throw an error if something went wrong with the request 1`] = `[Error: Error message]`;

exports[`Wrapper test Profile endpoints It should call the get profile attachments endpoint with the profile id 1`] = `
Object {
  "options": Object {
    "headers": Object {
      "X-API-KEY": "api_key",
      "X-USER-EMAIL": undefined,
    },
  },
  "url": Url {
    "auth": null,
    "hash": null,
    "host": "api.hrflow.ai",
    "hostname": "api.hrflow.ai",
    "href": "https://api.hrflow.ai/v1/profile/attachments?source_key=source_key&profile_key=profile_key",
    "path": "/v1/profile/attachments?source_key=source_key&profile_key=profile_key",
    "pathname": "/v1/profile/attachments",
    "port": null,
    "protocol": "https:",
    "query": "source_key=source_key&profile_key=profile_key",
    "search": "?source_key=source_key&profile_key=profile_key",
    "slashes": true,
  },
}
`;

exports[`Wrapper test Profile endpoints It should call the get profile document endpoint with the profile reference 1`] = `
Object {
  "options": Object {
    "headers": Object {
      "X-API-KEY": "api_key",
      "X-USER-EMAIL": undefined,
    },
  },
  "url": Url {
    "auth": null,
    "hash": null,
    "host": "api.hrflow.ai",
    "hostname": "api.hrflow.ai",
    "href": "https://api.hrflow.ai/v1/profile/attachments?source_key=source_key&reference=reference",
    "path": "/v1/profile/attachments?source_key=source_key&reference=reference",
    "pathname": "/v1/profile/attachments",
    "port": null,
    "protocol": "https:",
    "query": "source_key=source_key&reference=reference",
    "search": "?source_key=source_key&reference=reference",
    "slashes": true,
  },
}
`;

exports[`Wrapper test Profile endpoints It should call the get profile endpoint using Date number 1`] = `
Object {
  "options": Object {
    "headers": Object {
      "X-API-KEY": "api_key",
      "X-USER-EMAIL": undefined,
    },
  },
  "url": Url {
    "auth": null,
    "hash": null,
    "host": "api.hrflow.ai",
    "hostname": "api.hrflow.ai",
    "href": "https://api.hrflow.ai/v1/profiles/searching?source_keys=[%22source1%22,%22source2%22]&created_at_min=0&created_at_max=1514764800000&page=1&stage=YES&limit=30&sort_by=date_reception&order_by=desc",
    "path": "/v1/profiles/searching?source_keys=[%22source1%22,%22source2%22]&created_at_min=0&created_at_max=1514764800000&page=1&stage=YES&limit=30&sort_by=date_reception&order_by=desc",
    "pathname": "/v1/profiles/searching",
    "port": null,
    "protocol": "https:",
    "query": "source_keys=[%22source1%22,%22source2%22]&created_at_min=0&created_at_max=1514764800000&page=1&stage=YES&limit=30&sort_by=date_reception&order_by=desc",
    "search": "?source_keys=[%22source1%22,%22source2%22]&created_at_min=0&created_at_max=1514764800000&page=1&stage=YES&limit=30&sort_by=date_reception&order_by=desc",
    "slashes": true,
  },
}
`;

exports[`Wrapper test Profile endpoints It should call the get profile endpoint using Date object 1`] = `
Object {
  "options": Object {
    "headers": Object {
      "X-API-KEY": "api_key",
      "X-USER-EMAIL": undefined,
    },
  },
  "url": Url {
    "auth": null,
    "hash": null,
    "host": "api.hrflow.ai",
    "hostname": "api.hrflow.ai",
    "href": "https://api.hrflow.ai/v1/profiles/searching?source_keys=[%22source1%22,%22source2%22]&created_at_min=Thu%20Jan%2001%201970%2000:00:00%20GMT+0000%20(GMT)&created_at_max=Thu%20Jan%2001%201970%2000:00:01%20GMT+0000%20(GMT)&page=1&stage=YES&limit=30&sort_by=date_reception&order_by=desc",
    "path": "/v1/profiles/searching?source_keys=[%22source1%22,%22source2%22]&created_at_min=Thu%20Jan%2001%201970%2000:00:00%20GMT+0000%20(GMT)&created_at_max=Thu%20Jan%2001%201970%2000:00:01%20GMT+0000%20(GMT)&page=1&stage=YES&limit=30&sort_by=date_reception&order_by=desc",
    "pathname": "/v1/profiles/searching",
    "port": null,
    "protocol": "https:",
    "query": "source_keys=[%22source1%22,%22source2%22]&created_at_min=Thu%20Jan%2001%201970%2000:00:00%20GMT+0000%20(GMT)&created_at_max=Thu%20Jan%2001%201970%2000:00:01%20GMT+0000%20(GMT)&page=1&stage=YES&limit=30&sort_by=date_reception&order_by=desc",
    "search": "?source_keys=[%22source1%22,%22source2%22]&created_at_min=Thu%20Jan%2001%201970%2000:00:00%20GMT+0000%20(GMT)&created_at_max=Thu%20Jan%2001%201970%2000:00:01%20GMT+0000%20(GMT)&page=1&stage=YES&limit=30&sort_by=date_reception&order_by=desc",
    "slashes": true,
  },
}
`;

exports[`Wrapper test Profile endpoints It should call the get profile endpoint with the profile id 1`] = `
Object {
  "options": Object {
    "headers": Object {
      "X-API-KEY": "api_key",
      "X-USER-EMAIL": undefined,
    },
  },
  "url": Url {
    "auth": null,
    "hash": null,
    "host": "api.hrflow.ai",
    "hostname": "api.hrflow.ai",
    "href": "https://api.hrflow.ai/v1/profile?source_key=source_key&profile_key=profile_key",
    "path": "/v1/profile?source_key=source_key&profile_key=profile_key",
    "pathname": "/v1/profile",
    "port": null,
    "protocol": "https:",
    "query": "source_key=source_key&profile_key=profile_key",
    "search": "?source_key=source_key&profile_key=profile_key",
    "slashes": true,
  },
}
`;

exports[`Wrapper test Profile endpoints It should call the get profile endpoint with the profile reference 1`] = `
Object {
  "options": Object {
    "headers": Object {
      "X-API-KEY": "api_key",
      "X-USER-EMAIL": undefined,
    },
  },
  "url": Url {
    "auth": null,
    "hash": null,
    "host": "api.hrflow.ai",
    "hostname": "api.hrflow.ai",
    "href": "https://api.hrflow.ai/v1/profile?source_key=source_key&reference=reference",
    "path": "/v1/profile?source_key=source_key&reference=reference",
    "pathname": "/v1/profile",
    "port": null,
    "protocol": "https:",
    "query": "source_key=source_key&reference=reference",
    "search": "?source_key=source_key&reference=reference",
    "slashes": true,
  },
}
`;

exports[`Wrapper test Profile endpoints It should call the get profile parsing endpoint with the profile id 1`] = `
Object {
  "options": Object {
    "headers": Object {
      "X-API-KEY": "api_key",
      "X-USER-EMAIL": undefined,
    },
  },
  "url": Url {
    "auth": null,
    "hash": null,
    "host": "api.hrflow.ai",
    "hostname": "api.hrflow.ai",
    "href": "https://api.hrflow.ai/v1/profile/parsing?source_key=source_key&profile_key=profile_key",
    "path": "/v1/profile/parsing?source_key=source_key&profile_key=profile_key",
    "pathname": "/v1/profile/parsing",
    "port": null,
    "protocol": "https:",
    "query": "source_key=source_key&profile_key=profile_key",
    "search": "?source_key=source_key&profile_key=profile_key",
    "slashes": true,
  },
}
`;

exports[`Wrapper test Profile endpoints It should call the get profile parsing endpoint with the profile reference 1`] = `
Object {
  "options": Object {
    "headers": Object {
      "X-API-KEY": "api_key",
      "X-USER-EMAIL": undefined,
    },
  },
  "url": Url {
    "auth": null,
    "hash": null,
    "host": "api.hrflow.ai",
    "hostname": "api.hrflow.ai",
    "href": "https://api.hrflow.ai/v1/profile/parsing?source_key=source_key&reference=reference",
    "path": "/v1/profile/parsing?source_key=source_key&reference=reference",
    "pathname": "/v1/profile/parsing",
    "port": null,
    "protocol": "https:",
    "query": "source_key=source_key&reference=reference",
    "search": "?source_key=source_key&reference=reference",
    "slashes": true,
  },
}
`;

exports[`Wrapper test Profile endpoints It should call the get profile scoring endpoint with the profile id 1`] = `
Object {
  "options": Object {
    "headers": Object {
      "X-API-KEY": "api_key",
      "X-USER-EMAIL": undefined,
    },
  },
  "url": Url {
    "auth": null,
    "hash": null,
    "host": "api.hrflow.ai",
    "hostname": "api.hrflow.ai",
    "href": "https://api.hrflow.ai/v1/profiles/scoring?source_keys=[%22source_key%22]&job_key=job_key&board_key=board_key&use_agent=1",
    "path": "/v1/profiles/scoring?source_keys=[%22source_key%22]&job_key=job_key&board_key=board_key&use_agent=1",
    "pathname": "/v1/profiles/scoring",
    "port": null,
    "protocol": "https:",
    "query": "source_keys=[%22source_key%22]&job_key=job_key&board_key=board_key&use_agent=1",
    "search": "?source_keys=[%22source_key%22]&job_key=job_key&board_key=board_key&use_agent=1",
    "slashes": true,
  },
}
`;

exports[`Wrapper test Profile endpoints It should call the get profile scoring endpoint with the profile reference 1`] = `
Object {
  "options": Object {
    "headers": Object {
      "X-API-KEY": "api_key",
      "X-USER-EMAIL": undefined,
    },
  },
  "url": Url {
    "auth": null,
    "hash": null,
    "host": "api.hrflow.ai",
    "hostname": "api.hrflow.ai",
    "href": "https://api.hrflow.ai/v1/profiles/scoring?source_keys=[%22source_key%22]&job_key=job_key&board_key=board_key&use_agent=1",
    "path": "/v1/profiles/scoring?source_keys=[%22source_key%22]&job_key=job_key&board_key=board_key&use_agent=1",
    "pathname": "/v1/profiles/scoring",
    "port": null,
    "protocol": "https:",
    "query": "source_keys=[%22source_key%22]&job_key=job_key&board_key=board_key&use_agent=1",
    "search": "?source_keys=[%22source_key%22]&job_key=job_key&board_key=board_key&use_agent=1",
    "slashes": true,
  },
}
`;

exports[`Wrapper test Profile endpoints It should call the post profile data endpoint 1`] = `
Object {
  "options": Object {
    "headers": Object {
      "X-API-KEY": "api_key",
      "X-USER-EMAIL": undefined,
    },
    "method": "POST",
  },
  "url": Url {
    "auth": null,
    "hash": null,
    "host": "api.hrflow.ai",
    "hostname": "api.hrflow.ai",
    "href": "https://api.hrflow.ai/v1/profile/indexing",
    "path": "/v1/profile/indexing",
    "pathname": "/v1/profile/indexing",
    "port": null,
    "protocol": "https:",
    "query": null,
    "search": null,
    "slashes": true,
  },
}
`;

exports[`Wrapper test Source endpoints It should call the get source endpoint 1`] = `
Object {
  "options": Object {
    "headers": Object {
      "X-API-KEY": "api_key",
      "X-USER-EMAIL": undefined,
    },
  },
  "url": Url {
    "auth": null,
    "hash": null,
    "host": "api.hrflow.ai",
    "hostname": "api.hrflow.ai",
    "href": "https://api.hrflow.ai/v1/source?source_key=source_key",
    "path": "/v1/source?source_key=source_key",
    "pathname": "/v1/source",
    "port": null,
    "protocol": "https:",
    "query": "source_key=source_key",
    "search": "?source_key=source_key",
    "slashes": true,
  },
}
`;

exports[`Wrapper test Source endpoints It should call the get source list endpoint 1`] = `
Object {
  "options": Object {
    "headers": Object {
      "X-API-KEY": "api_key",
      "X-USER-EMAIL": undefined,
    },
  },
  "url": Url {
    "auth": null,
    "hash": null,
    "host": "api.hrflow.ai",
    "hostname": "api.hrflow.ai",
    "href": "https://api.hrflow.ai/v1/sources",
    "path": "/v1/sources",
    "pathname": "/v1/sources",
    "port": null,
    "protocol": "https:",
    "query": null,
    "search": null,
    "slashes": true,
  },
}
`;
